### PR TITLE
Preliminary work to removing GtkMenu 

### PR DESCRIPTION
**Pull request by bhennion**
This PR is the first part of an attempt to split #4372 into reviewable PR's.

Here are just preliminary inoccuous changes (unused variables, const-correctness and smart pointers in gui/)

### CODE REVIEW

**Review by Technius — APPROVED** [review]

**Comment by bhennion** [comment]
Merging in 24 hours unless an objection is raised.

**Comment by bhennion** [comment]
I'll wait until #4455 is merged before merging this.

**Pull request merged**
